gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -909,48 +1005,133 @@ gst_tensor_filter_transform (GstBaseTransform * trans,
       meta.format = _NNS_TENSOR_FORMAT_FLEXIBLE;
 
       flex_mem = gst_memory_new_wrapped (0,
-          out_tensors[i].data, out_tensors[i].size, 0,
-          out_tensors[i].size, out_tensors[i].data, g_free);
+          out_trans_data->tensors[i].data, out_trans_data->tensors[i].size, 0,
+          out_trans_data->tensors[i].size, out_trans_data->tensors[i].data,
+          g_free);
 
-      out_mem[i] = gst_tensor_meta_info_append_header (&meta, flex_mem);
+      out_trans_data->mem[i] =
+          gst_tensor_meta_info_append_header (&meta, flex_mem);
       gst_memory_unref (flex_mem);
-    } else if (allocate_in_invoke) {
+    } else if (out_trans_data->allocate_in_invoke) {
       /* prepare memory block if successfully done */
-      out_mem[i] = mem = gst_tensor_filter_get_wrapped_mem (self,
-          out_tensors[i].data, out_tensors[i].size);
+      out_trans_data->mem[i] = mem = gst_tensor_filter_get_wrapped_mem (self,
+          out_trans_data->tensors[i].data, out_trans_data->tensors[i].size);
 
-      if (out_flexible) {
+      if (out_trans_data->is_flexible) {
         /* prepare new memory block with meta */
-        out_mem[i] = gst_tensor_meta_info_append_header (&out_meta[i], mem);
+        out_trans_data->mem[i] =
+            gst_tensor_meta_info_append_header (&out_trans_data->meta[i], mem);
         gst_memory_unref (mem);
       }
     }
 
     /* append the memory block to outbuf */
-    gst_tensor_buffer_append_memory (outbuf, out_mem[i],
+    gst_tensor_buffer_append_memory (outbuf, out_trans_data->mem[i],
         gst_tensors_info_get_nth_info (&prop->output_meta, i));
   }
+}
 
-  return GST_FLOW_OK;
+/**
+ * @brief non-ip transform. required vmethod of GstBaseTransform.
+ */
+static GstFlowReturn
+gst_tensor_filter_transform (GstBaseTransform * trans,
+    GstBuffer * inbuf, GstBuffer * outbuf)
+{
+  GstTensorFilter *self = GST_TENSOR_FILTER_CAST (trans);
+  GstTensorFilterPrivate *priv = &self->priv;
+  GstTensorFilterProperties *prop = &priv->prop;
+  guint i;
+  gint invoke_res = -1;
+  gboolean need_profiling;
+  GstFlowReturn retval = GST_FLOW_OK;
+  FilterTransformData *in_trans_data = NULL;
+  FilterTransformData *out_trans_data = NULL;
+  GstTensorMemory *invoke_tensors = NULL;
+
+  /* 0. Check all properties. */
+  retval = _gst_tensor_filter_transform_validate (trans, inbuf, outbuf);
+  if (retval != GST_FLOW_OK)
+    return retval;
+
+  /* 1. Get all input tensors from inbuf. */
+  in_trans_data = _gst_tensor_filter_transform_get_input_data (trans, inbuf);
+  if (!in_trans_data) {
+    return GST_FLOW_ERROR;
+  }
+
+  invoke_tensors =
+      _gst_tensor_filter_transform_get_invoke_tensors (trans, in_trans_data);
+  if (!invoke_tensors) {
+    goto mem_map_error;
+  }
+
+  /* 2. Prepare output tensors. */
```

Comment:
Let's remove comment, how about you?

Reply:
Ok, The function names are enough to explain the its operation. Removed.